core/src/main/java/com/linecorp/armeria/server/throttling/RequestThrottlingStrategy.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2017 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server.throttling;
+
+import java.util.Random;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.function.Supplier;
+
+import com.linecorp.armeria.common.Request;
+
+/**
+ * Determines whether a failed request should be retried.
+ */
+@FunctionalInterface
+public interface RequestThrottlingStrategy {
+    /**
+     * A {@link RequestThrottlingStrategy} that defines a throttling should not be performed.
+     */
+    RequestThrottlingStrategy NEVER = request -> false;
+
+    /**
+     * Returns a {@link RequestThrottlingStrategy} that throttles {@code rate} percent of incoming requests.
+     */
+    static RequestThrottlingStrategy rateLimit(float rate) {
+        return rateLimit(rate, ThreadLocalRandom::current);
+    }
+
+    /**
+     * Returns a {@link RequestThrottlingStrategy} that throttles {@code rate} percent of incoming requests.
+     */
+    static RequestThrottlingStrategy rateLimit(float rate, Supplier<Random> randomSupplier) {
```

Comment:
I'm not sure this will ever be used directly by a user. What a user wants would be to create a `RequestThrottlingStrategy` which can be turned on/off dynamically.